src/Compilers/CSharp/Portable/Symbols/TypeUnification.cs

```diff
@@ -19,28 +17,28 @@ internal static class TypeUnification
         /// </summary>
         public static bool CanUnify(TypeSymbol t1, TypeSymbol t2)
         {
-            if (TypeSymbol.Equals(t1, t2, TypeCompareKind.ConsiderEverything2))
+            if (TypeSymbol.Equals(t1, t2, TypeCompareKind.CLRSignatureCompareOptions))
```

Comment:
>CLRSignatureCompareOptions [](start = 58, length = 26)

It feels like we need to have a test for nullable since this change is likely to have effect for it.

Reply:
Is this covered in TypeUnification_02?

---
In reply to: [501408203](https://github.com/dotnet/roslyn/pull/48349#discussion_r501408203) [](ancestors = 501408203)

Reply:
Yes, this is covered in `TypeUnification_02`.

---
In reply to: [501411969](https://github.com/dotnet/roslyn/pull/48349#discussion_r501411969) [](ancestors = 501411969,501408203)

Reply:
> Yes, this is covered in TypeUnification_02.

I think we should have a test that observes expected diagnostics.

---
In reply to: [501966682](https://github.com/dotnet/roslyn/pull/48349#discussion_r501966682) [](ancestors = 501966682,501411969,501408203)

Reply:
There is only one caller, `SourceMemberContainerSymbol.CheckInterfaceUnification()`, and that caller only calls `CanUnify()` with distinct interfaces.

---
In reply to: [502147027](https://github.com/dotnet/roslyn/pull/48349#discussion_r502147027) [](ancestors = 502147027,501966682,501411969,501408203)

Reply:
> There is only one caller, SourceMemberContainerSymbol.CheckInterfaceUnification(), and that caller only calls CanUnify() with distinct interfaces.

Consider adding a test that verifies user-observable compiler behavior

---
In reply to: [505159752](https://github.com/dotnet/roslyn/pull/48349#discussion_r505159752) [](ancestors = 505159752,502147027,501966682,501411969,501408203)